app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,21 @@
+class CommentsController < ApplicationController
+  def create
+    @post = Post.find(params[:post_id])
+    @user = @post.author
+    @comment = @post.comments.build(comment_params)
+    @comment.user = current_user
+
+    if @comment.save
+      flash[:success] = 'Comment created successfully.'
+      redirect_to user_post_path(@user, @post)
+    else
+      flash[:error] = 'Comment could not be created.'
+    end
+  end
+
```

Comment:
- [ ] In your comments controller It would be nice to add the `new` action as this is missing in your controller, see the example below ⬇️  

```
  def new
    @comment = Comment.new
  end
```

Reply:
This change has not been done well